src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1066,25 +1066,16 @@ internal unsafe int IndexOf(string source, string value, int startIndex, int cou
                 *matchLengthPtr = 0;
             }
 
-            if (source.Length == 0)
+            if (value.Length == 0)
```

Comment:
2 callers expect `value.Length == 0` to be tested first
3rd caller `ReplaceCore` doesn't allow a `value.Length == 0` so moving it first is ok